openfl-tutorials/experimental/Workflow_FedProx_Keras_MNIST.ipynb

```diff
@@ -0,0 +1,1304 @@
+   "metadata": {},
+   "source": [
+    "This tutorial demonstrates how to train a model using TensorFlow Keras on the MNIST dataset, leveraging the new FedAI Workflow Interface. The Workflow Interface provides a novel way to compose federated learning experiments with OpenFL, enabling researchers to handle non-IID data and perform federated averaging with optimizations like FedProx. Through this tutorial, you will learn how to set up the federated learning environment, define the flow, and execute the training process across multiple collaborators.\n"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "39c3d86a",
+   "metadata": {},
+   "source": [
+    "# What is it?"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "a7989e72",
+   "metadata": {},
+   "source": [
+    "The workflow interface is a new way of composing federated learning expermients with OpenFL. It was borne through conversations with researchers and existing users who had novel use cases that didn't quite fit the standard horizontal federated learning paradigm. "
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "fc8e35da",
+   "metadata": {},
+   "source": [
+    "# Getting Started"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "4dbb89b6",
+   "metadata": {},
+   "source": [
+    "First we start by installing the necessary dependencies for the workflow interface"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "f7f98600",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%pip install tensorflow\n",
+    "%pip install openfl\n",
+    "%pip install -r workflow_interface_requirements.txt"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b6fecf9d",
+   "metadata": {},
+   "source": [
+    "Now, we import the relevant modules and do some basic initializations  "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "88fdb191",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "import tensorflow as tf\n",
+    "\n",
+    "tf.config.run_functions_eagerly(True)\n",
+    "tf.random.set_seed(0)\n",
+    "np.random.seed(0)\n",
+    "\n",
+    "def test_intel_tensorflow():\n",
+    "    \"\"\"\n",
+    "    Check if Intel version of TensorFlow is installed\n",
+    "    \"\"\"\n",
+    "    import tensorflow as tf\n",
+    "\n",
+    "    print(\"We are using Tensorflow version {}\".format(tf.__version__))\n",
+    "\n",
+    "    major_version = int(tf.__version__.split(\".\")[0])\n",
+    "    if major_version >= 2:\n",
+    "        from tensorflow.python.util import _pywrap_util_port\n",
+    "        print(\"Intel-optimizations (DNNL) enabled:\",\n",
+    "              _pywrap_util_port.IsMklEnabled())\n",
+    "    else:\n",
+    "        print(\"Intel-optimizations (DNNL) enabled:\")\n",
+    "\n",
+    "test_intel_tensorflow()"
```

Comment:
Not required. Unnecessary cognitive overhead + all tensorflow binaries are Intel-optimized by default.

Reply:
fixed